### PR TITLE
Feature/sn 539 edit entity

### PR DESCRIPTION
**Pull request by AlexanderMakarov124**

### CODE REVIEW

File: src/Saritasa.NetForge.Blazor/Controls/CustomFields/BoolField.razor.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.AspNetCore.Components;
+using Saritasa.NetForge.UseCases.Metadata.GetEntityById;
+
+namespace Saritasa.NetForge.Blazor.Controls.CustomFields;
+
+/// <summary>
+/// Represents bool field.
+/// </summary>
+public partial class BoolField
```
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.Blazor/Controls/CustomFields/BoolField.razor.cs:9]: Can you explain to me what is this class responsible for?

File: src/Saritasa.NetForge.Blazor/Controls/CustomFields/BoolField.razor.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    [Parameter]
+    [EditorRequired]
+    public object EntityModel { get; init; } = null!;
+
+    [Parameter]
+    [EditorRequired]
+    /// <summary>
+    /// Is field with read only access.
+    /// </summary>
+    public bool IsReadOnly { get; init; }
```
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.Blazor/Controls/CustomFields/BoolField.razor.cs:30]: Please move the comment above the attributes so it doesn't give us a warning.

File: src/Saritasa.NetForge.Blazor/Controls/CustomFields/DateField.razor.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    [Parameter]
+    [EditorRequired]
+    public object EntityModel { get; init; } = null!;
+
+    [Parameter]
+    [EditorRequired]
+    /// <summary>
+    /// Is field with read only access.
+    /// </summary>
+    public bool IsReadOnly { get; init; }
```
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.Blazor/Controls/CustomFields/DateField.razor.cs:30]: Move the comment of this property too.

File: src/Saritasa.NetForge.Blazor/Controls/CustomFields/EnumField.razor.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    [Parameter]
+    [EditorRequired]
+    public object EntityModel { get; init; } = null!;
+
+    [Parameter]
+    [EditorRequired]
+    /// <summary>
+    /// Is field with read only access.
+    /// </summary>
+    public bool IsReadOnly { get; init; }
```
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.Blazor/Controls/CustomFields/EnumField.razor.cs:30]: The same as above.

File: src/Saritasa.NetForge.Blazor/Controls/CustomFields/NumberField.razor.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    [Parameter]
+    [EditorRequired]
+    public object EntityModel { get; init; } = null!;
+
+    [Parameter]
+    [EditorRequired]
+    /// <summary>
+    /// Is field with read only access.
+    /// </summary>
+    public bool IsReadOnly { get; init; }
```
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.Blazor/Controls/CustomFields/NumberField.razor.cs:30]: The same as above.

File: src/Saritasa.NetForge.Infrastructure.EfCore/Services/EfCoreDataService.cs
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.Infrastructure.EfCore/Services/EfCoreDataService.cs:1]: Please fix these warnings.
image

File: src/Saritasa.NetForge.UseCases/Metadata/GetEntityById/PropertyMetadataDto.cs [hunk 6]
```diff
@@ -20,6 +20,9 @@ public record PropertyMetadataDto
     /// <inheritdoc cref="PropertyMetadata.IsPrimaryKey"/>
     public bool IsPrimaryKey { get; set; }
 
+    /// <inheritdoc cref="PropertyMetadata.ClrType"/>
```
Comment (toanphamsaritasa) [inline src/Saritasa.NetForge.UseCases/Metadata/GetEntityById/PropertyMetadataDto.cs:23]: Fix these warnings too.
image

File: src/Saritasa.NetForge.Blazor/Controls/CustomFields/BoolField.razor.cs (hunk 1 above)
Comment (AlexanderMakarov124) [reply src/Saritasa.NetForge.Blazor/Controls/CustomFields/BoolField.razor.cs:9]: We have this code on `CreateEntity` and `EditEntity` pages
```csharp
var componentType = CustomFieldHelper.GetComponentType(property.ClrType!);
var parameters = new Dictionary<string, object>
{
    { "Property", property },
    { "EntityModel", ViewModel.EntityModel },
    { "IsReadOnly", property.IsPrimaryKey }
};

<DynamicComponent Type="componentType" Parameters="parameters" />
```

 If `property.ClrType` has `bool` type, then `DynamicComponent` will render this `BoolField`. The same situation with others fields

File: src/Saritasa.NetForge.Blazor/Pages/EditEntity.razor [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    var title = $"Edit {ViewModel.Model.DisplayName.ToMeaningfulName()}";
+    <PageTitle>@title</PageTitle>
+    <MudText Class="pb-4" Typo="Typo.h4">@title</MudText>
+
+    <MudGrid Justify="Justify.Center">
+        <MudItem xs="6">
+            <EditForm EditContext="new EditContext(ViewModel.EntityModel)" OnValidSubmit="ViewModel.UpdateEntityAsync">
+                <div class="d-flex flex-column flex-grow-1 gap-4">
+                    @{
+                        var properties = ViewModel.Model.Properties
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EditEntity.razor:25]: Can we apply this filtering from the ViewModel side?

File: src/Saritasa.NetForge.Blazor/Pages/EditEntity.razor.cs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    private AdminOptions? AdminOptions { get; set; }
+
+    /// <summary>
+    /// Entity id.
+    /// </summary>
+    [Parameter]
+    public string StringId { get; init; } = null!;
+
+    /// <summary>
+    /// Entity instance id.
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EditEntity.razor.cs:29]: Primary key?

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs [hunk 9]
```diff
@@ -10,7 +11,7 @@ namespace Saritasa.NetForge.Blazor.Pages;
 /// Entity details.
 /// </summary>
 [Route("/entities/{stringId}")]
-public partial class EntityDetails : MvvmComponentBase<EntityDetailsViewModel>
+public partial class EntityDetails : MvvmComponentBase<EntityDetailsViewModel>, IDisposable
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:14]: Please, fix the warnings

File: src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs [hunk 10]
```diff
@@ -45,4 +46,13 @@ private void NavigateToCreation()
...
         NavigationService.NavigateTo<CreateEntityViewModel>(parameters: StringId);
     }
+
+    private void NavigateToEditing(DataGridRowClickEventArgs<object> row)
+    {
+        var primaryKeyValues = ViewModel.Model.Properties
+            .Where(property => property.IsPrimaryKey)
+            .Select(primaryKey => ViewModel.GetPropertyValue(row.Item, primaryKey).ToString()!);
+
+        NavigationService.NavigateTo<EditEntityViewModel>(parameters: [StringId, string.Join("--", primaryKeyValues)]);
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Blazor/Pages/EntityDetails.razor.cs:56]: Fix the warnings

**Review by vasiliy-chefonov — APPROVED** [review]

**Review by toanphamsaritasa — APPROVED** [review]

**Pull request merged**